### PR TITLE
Unified output handling through Pharos::Output

### PR DESCRIPTION
**Pull request by kke**

### CODE REVIEW

**Comment by kke** [comment]
The reason to use severity 6 for top level messages is to optionally reduce the noise from phases.

image

```diff
diff --git a/lib/pharos/command.rb b/lib/pharos/command.rb
index ee345da..3850324 100644
--- a/lib/pharos/command.rb
+++ b/lib/pharos/command.rb
@@ -14,5 +14,9 @@ module Pharos
     option ['-d', '--debug'], :flag, "enable debug output", environment_variable: "DEBUG" do
       Out.debug = true
     end
+
+    option ['-V', '--verbose'], :flag, "enable verbose output" do
+      Out.verbose = true
+    end
   end
 end
diff --git a/lib/pharos/output.rb b/lib/pharos/output.rb
index 897e472..1675f9e 100644
--- a/lib/pharos/output.rb
+++ b/lib/pharos/output.rb
@@ -24,7 +24,7 @@ module Pharos

     def configure_logger
       @logger = Logger.new($stdout)
-      @logger.level = Logger::INFO
+      @logger.level = Logger::WARN
       @logger.formatter = STDOUT_FORMATTER if $stdout.tty?
     end

@@ -56,7 +56,13 @@ module Pharos
     # Set debug on/off
     # @param [TrueClass,FalseClass]
     def debug=(bool)
-      @logger.level = bool ? Logger::DEBUG : Logger::INFO
+      @logger.level = bool ? Logger::DEBUG : Logger::WARN
+    end
+
+    # Set verbose on/off
+    # @param [TrueClass,FalseClass]
+    def verbose=(bool)
+      @logger.level = bool ? Logger::INFO : Logger::WARN
     end

     # Set color on/off
```

**Comment by kke** [comment]
Yes, looks like the host got lost from the phase messages.

I assume the person running the cluster has some kind of estimate of how many nodes there are. Could just be "Fooing bar on worker nodes".

File: lib/pharos/phases/configure_master.rb [hunk 1]
```diff
@@ -59,16 +59,17 @@ def install
         push_authentication_token_webhook_config if @config.authentication&.token_webhook
 
         copy_kube_certs
+        configure_secrets_encryption
```
Comment (SpComb) [inline lib/pharos/phases/configure_master.rb]: mergefail

File: lib/pharos/phases/configure_secrets_encryption.rb [hunk 2]
```diff
@@ -21,11 +21,11 @@ def call
 
       # @return [Hash, nil]
       def read_config_keys
-        logger.debug { "Checking if secrets encryption is already configured ..." }
+        Out.debug(@host.address) { "Checking if secrets encryption is already configured ..." }
```
Comment (SpComb) [inline lib/pharos/phases/configure_secrets_encryption.rb]: Should there also be `debug`, `warn`, `error` helpers on the `Pharos::Phase`?

**Comment by kke** [comment]
Regular:

image

Verbose:

image

**Comment by kke** [comment]
Seriously outdated

**Pull request closed without merging**